framework/src/base/FEProblem.C

```diff
@@ -1882,6 +1898,31 @@ FEProblem::reinitMaterials(SubdomainID blk_id, THREAD_ID tid, bool swap_stateful
 }
 
 void
+FEProblem::reinitMaterialsDirac(SubdomainID blk_id, THREAD_ID tid, bool swap_stateful)
+{
+  if (_all_materials[Moose::DIRAC_MATERIAL_DATA].hasActiveBlockObjects(blk_id, tid))
+  {
+    const Elem * & elem = _assembly[tid]->elem();
+    unsigned int n_points = _assembly[tid]->qRule()->n_points();
+
+    if (_dirac_material_data[tid]->nQPoints() != n_points)
```

Comment:
That is correct. Currently I do not have a good solution for that.

I understand that the ```MaterialPropertyStorage``` does not store any information about the quadrature coordinates but just the quadrature index. So the book keeping will be complicated.

Moving a point within a element is one situation. Another is to add and delete the same number of points. For that case, the return value is not correct neither. :-(